torch/csrc/jit/frontend/concrete_module_type.cpp

```diff
@@ -269,9 +297,10 @@ void ConcreteModuleTypeBuilder::addBuiltinFunction(
 
 void ConcreteModuleTypeBuilder::addModule(
     std::string name,
-    std::shared_ptr<ConcreteModuleType> meta) {
-  modules_.emplace_back(
-      ConcreteModuleTypeBuilder::ModuleInfo{std::move(name), std::move(meta)});
+    std::shared_ptr<ConcreteModuleType> meta,
```

Comment:
The api here is seems to be a little off from everything else, where we exclusively use the `set{...}` functions. Is there a reason for the divergence ?

Reply:
Yeah, so in accordance with considering the annotation as a property of the parent and not of the submodule to which it applies, the annotation is added here, in `addModule` rather than directly on the parent. We could add it later using something like `ConcreteModuleTypeBuilder ::setSubmoduleContainedTypeHint(name, hint)` but it seems like an unnecessary second function call. I don't feel too strongly about this though so I'd be happy to make this change.